### PR TITLE
285 history button

### LINKED ISSUE #285: `History` button for all resources
There is value in seeing the audit history for all kinds of items.

For `Items` we've put the history on the right-hand side of the `ItemShow` component.  But, we can't afford to do this for all resource types.

So, instead we should create a `ResourceHistory` component. This will take a resource name and id.  On each resource we'll add a `History` icon action to the `Show` page (using the `history` icon from material icons).  

It will popup an `AuditList` table showing the history for that resource item. This will have a `Print` button.

We can then remove the `AuditList` from `ItemShow`.

### PR DESCRIPTION
**Pull request by TahaKhanAbdalli**
Fixes #285

### CODE REVIEW

**Comment by IanMayo** [comment]
(I've lost this message, so re-typing)

Could we also have the `history` button for `Users` and all of the `Reference Data` resources, please?

**Comment by IanMayo** [comment]
Thanks @TahaKhanAbdalli - but could you solve these issues please?

(I know it's just the linter, and the code works fine, but I'd like to keep us at zero linter issues ;-) )
image

**Review by IanMayo — CHANGES_REQUESTED** [review]
I've just logged in as `ian` in dev-tools I can see that the audit item has been created. But, when I click `History` for the user `ian` it is empty.  This looks like a bug.

**Comment by TahaKhanAbdalli** [comment]
The `login` and `logout` actions do not have any `resource` attached to them, therefore the `history` shows empty. But if you depart user or do any actions whose `target resource` is users, then it will show in the history page with items listed.

**Comment by IanMayo** [comment]
Sure. The history for `UserShow` is a special case. It shows edits to that resource.  It should also include data changes conducted by that resource (including login, loan, edit).

This will let us view/review all of the interactions for that RCO user.

**Comment by IanMayo** [comment]
Hello @TahaKhanAbdalli - is it possible to show user events that _either_ record changes to the details for this user, _or_ changes performed by this user, _or_ activities that relate to this user (items loaned to them).

File: src/resources/vault-locations/VaultLocationList.tsx [hunk 1]
```diff
@@ -21,10 +21,6 @@ export default function VaultLocationList(): React.ReactElement {
     [record]
   )
```
Comment (IanMayo) [inline src/resources/vault-locations/VaultLocationList.tsx:23]: Hello @TahaKhanAbdalli  - I've tried calling the `setOpen` state entity directly from the TSX.

It simplifies the code, but is it ok to do that?

**Review by IanMayo — APPROVED** [review]

**Pull request merged**